shared/src/main/java/com/faforever/neroxis/util/ImageUtil.java

```diff
@@ -171,23 +171,32 @@ public static void writeRawDDS(BufferedImage image, Path path) throws IOExceptio
         Files.write(path, imageBytes.array(), StandardOpenOption.APPEND);
     }
 
-    public static BufferedImage getMapwideTexture(NormalMask normalMask, FloatMask waterDepth, FloatMask shadowMask) {
-        if (shadowMask.getSize() != normalMask.getSize()) {
-            throw new IllegalArgumentException("Mask sizes do not match: shadow size %d, normal size %d"
-                                                       .formatted(shadowMask.getSize(), normalMask.getSize()));
-        }
+    public static BufferedImage getMapInfoTexture(FloatMask waterDepth, FloatMask shadowMask) {
         waterDepth.resample(shadowMask.getSize());
```

Comment:
This should be using a copy of waterDepth not changing the size of the passed in mask if it needs to be rescaled just for this operation

Reply:
because it would also change the original floatmask even out of the scope of this function, right?

Reply:
Yes